### PR TITLE
Fix buffer overflow #17122

### LINKED ISSUE #17122: Stack Smashing with less (~...) when the input buffer > 0x200
### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | Archlinux 64
| File format of the file you reverse (mandatory)      | ELF
| Architecture/bits of the file (mandatory)            | x86/64
| r2 -v full output, **not truncated** (mandatory)     | radare2 4.5.0-git 24819 @ linux-x86-64 git.4.4.0-322-g5124bef43 commit: 5124bef4343d114a1e781aaa79aca3a36bcbf713 build: 2020-06-22__09:27:13

### Expected behavior
No Stack Mash

### Actual behavior
2020-06-22-093059_404x62_scrot

2020-06-22-093658_714x561_scrot

### Steps to reproduce the behavior 
```
> aaa
> afl~...
> # (Move the mouse wheel until there is so much garbage char that you can't see the input line and press CTRL-C, it complicated to reproduced every time)
```

### Additional Logs, screenshots, source-code,  configuration dump, ...
[bin.zip](https://github.com/radareorg/radare2/files/4811808/bin.zip)

### PR DESCRIPTION
**Pull request by 08A**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**
Draft of a possible solution. I will push an other implementation with a buffer.

**Test plan**
see #17122

**Closing issues**
close #17122

### CODE REVIEW

**Comment by 08A** [comment]
```
257                 int i;
258                 for (i = I(line)->buffer.length; i > I(line)->buffer.index; i--) {
259                         hud_prompt[i] = hud_prompt[i - 1];
260                 }
```

There is no guarantee that i < HUD_BUF_SIZE + 1
And i don't understant why we use `hud_prompt`.
Imho we could remove the buffer and use `user_input`

**Comment by ret2libc** [comment]
I don't understand it either... IMHO you can remove it and use user_input, with the addition of just printing `|` instead of putting it in the buffer.

File: libr/cons/hud.c [hunk 1]
```diff
@@ -250,15 +249,9 @@ R_API char *r_cons_hud(RList *list, const char *prompt) {
 #endif
 		r_cons_visual_flush ();
 		(void) r_line_readline ();
-		memset (user_input, 0, HUD_BUF_SIZE);
-		memset (hud_prompt, 0, HUD_BUF_SIZE + 1);
 		strncpy (user_input, I(line)->buffer.data, HUD_BUF_SIZE - 1); 				// to search
```
Comment (ret2libc) [inline libr/cons/hud.c:252]: this should be `HUD_BUF_SIZE`.

**Comment by 08A** [comment]
I moved the `user_input[HUD_PROMPT_SIZE] = 0` outside the loop, because the last byte is never override.

**Pull request merged**